### DISCUSSION TITLE
Is application specific keybinding possible?

### DISCUSSION THREAD

**Discussion by Momijie** [i3/i3#5078] [Q&A] [answered]
I am wondering if it is possible to assign a key binding to a specific application? By that I mean a key will act by its default when not in focus of an application that changes its behavior. 

I got the first part of the desired effect to work here with this script using xdotool
```
#!/bin/bash

title=$(xdotool getwindowclassname $(xdotool getwindowfocus))

# Application specific check
if [ "$title" = "$1" ]; then
    # send the hotkey
    xdotool key $2
else
    # send the key that was originally pressed
    # xdotool key $3
    # but this triggers a loop because of i3 detecting the key press again, I assume
fi
```
And I assign it to the desired key with
`bindsym --release F1 exec ~/.scripts/override librewolf Control+Page_Up F1`

**Comment by Airblader** [comment, marked as answer]
No, i3 does not support that. There may be external apps that support this, though, but I don't know them.
